pkg/gadgets/network-graph/tracer/tracer.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/cilium/ebpf"
+	"golang.org/x/sys/unix"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/rawsock"
+)
+
+//go:generate bash -c "source ./clangosflags.sh; go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang graphmap ./bpf/graphmap.c -- $CLANG_OS_FLAGS -I./bpf/"
+
+//go:generate bash -c "source ./clangosflags.sh; go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang graph ./bpf/graph.c -- $CLANG_OS_FLAGS -I./bpf/"
+
+// /* for htons() and htonl() */
+// #include <arpa/inet.h>
+import "C"
+
+const (
+	BPFSocketAttach = 50
+)
+
+type Edge struct {
+	Key     string
+	PktType string
+	IP      net.IP
+	Proto   string
+	Port    int
+}
+
+type link struct {
+	networkGraphObjects graphObjects
+
+	sockFd int
+
+	containerQuark uint64
+
+	// users count how many users called Attach(). This can happen for two reasons:
+	// 1. several containers in a pod (sharing the netns)
+	// 2. pods with networkHost=true
+	users int
+}
+
+type Tracer struct {
+	config *Config
```

Comment:
Is this field being used?

Reply:
No, at the moment, the `Config struct` is empty. Should I remove it now and add it when/if we need it? (That would change the API of the `tracer` pkg, but nothing relies on it now)